### PR TITLE
Fix and enable cutting out of sub-grid pieces.

### PR DESCRIPTION
**Pull request by rbrune**
My little project required an easy way to arbitrarily cut a bigger map into sub-tiles. Introducing new command line arguments and fixing up the slicing code makes this super easy.

### CODE REVIEW

File: lidartile/grid.py [hunk 1]
```diff
@@ -50,16 +50,16 @@ def smooth(self, factor=1):
...
-        assert x + w <= self.width
-        assert y + h <= self.height
+        assert ox + w <= self.width
+        assert oy + h <= self.height
         result = self.__class__(w, h)
-        for y in range(y, y + h):
-            idx = x + (y * w)
-            ouridx = x + (y * self.width)
+        for y in range(0, h):
+            idx = 0 + (y * w)
```
Comment (andrewgodwin) [inline lidartile/grid.py]: I'm not sure you need `0 +` here?

**Review by andrewgodwin — COMMENTED** [review]
Very close to perfect, one tiny change! And thanks!

**Comment by rbrune** [comment]
Indeed! Let me fix this.

**Review by andrewgodwin — APPROVED** [review]

**Comment by mert-ozdemirr** [comment]
I have no idea about how to run your program I have .asc files but I don't know how to convert them to 3-d models could you help me with that?

**Pull request merged**